packages/browser-destinations/src/destinations/heap/trackEvent/index.ts

```diff
@@ -56,20 +56,59 @@ const action: BrowserActionDefinition<Settings, HeapApi, Payload> = {
     }
   },
   perform: (heap, event) => {
-    const eventProperties = Object.assign({}, event.payload.properties)
-    eventProperties.segment_library = HEAP_SEGMENT_BROWSER_LIBRARY_NAME
+    // Add user properties
     if (event.payload.anonymousId || isDefined(event.payload?.traits)) {
       const traits = flat(event.payload?.traits)
       heap.addUserProperties({
         ...(isDefined(event.payload.anonymousId) && { anonymous_id: event.payload.anonymousId }),
         ...(isDefined(traits) && traits)
       })
     }
+
+    // Identify user
     if (event.payload?.identity && isDefined(event.payload?.identity)) {
       heap.identify(event.payload.identity)
     }
-    heap.track(event.payload.name, eventProperties)
+
+    // Track Events
+    const eventProperties = Object.assign({}, event.payload.properties)
+    const eventName = event.payload.name
+    const arrayProperties: { [key: string]: Properties[] } = {}
+
+    for (const [key, value] of Object.entries(eventProperties)) {
+      if (Array.isArray(value)) {
+        arrayProperties[key] = value
+        delete eventProperties[key]
+      }
+    }
+
+    heapTrack(heap, eventName, eventProperties)
+
+    for (const [arrayPropertyKey, arrayPropertyValues] of Object.entries(arrayProperties)) {
+      arrayPropertyValues.forEach((arrayPropertyValue) => {
+        let arrayProperties = {}
+        for (const [key, value] of Object.entries(arrayPropertyValue)) {
+          if (typeof value == 'object' && value !== null) {
+            arrayProperties = { ...arrayProperties, ...flat(value as Properties, key) }
+          } else {
+            arrayProperties = Object.assign(arrayProperties, { [key]: value })
+          }
+        }
+        heapTrack(heap, `${eventName} ${arrayPropertyKey} item`, arrayProperties)
```

Comment:
If you have multiple arrays in the properties blob, what do we want to call them? Thought it might be good to include the name in that too...

Reply:
great idea!